### DISCUSSION TITLE
Simple text report?

### DISCUSSION THREAD

**Discussion by tdgvrr** [allinurl/goaccess#2256] [Q&A]
Is there a way to produce a simple text file containing the data I see on the console? 

Interactive usage with ncurses is great, but I need to capture the statistics I see on the screen in plain text. HTML, JSON and CSV aren't much help for me - all I want is the console display with all the sections expanded, then redirected to an output text file.

**Comment by allinurl** [comment]
Currently there's no option to output in plain text. The closest to this request will be issue #685. i wonder, are you looking to email the plain text report?

**Comment by tdgvrr** [comment]
Thanks for the reply!

As you know, the Apache access logs are a goldmine of useful information,
and your app is very handy at summarizing this data. We're looking to do a
few different things with the data you produce:

   1. Anomaly detection: run the report on a scheduled basis (hourly?) and
   compare metrics from one interval to the next, generating alerts for any
   unusual activity.
   2. SLA's: we have various service level agreements with our stakeholders
   - we'd like to have an automated way of proving that we're meeting SLAs.
   3. Security/intrusion detection: join your report with other
   information coming from our security logs to better understand attack
   patterns.
   4. "Service Status" web pages: like a lot of websites, we have a page
   showing the status of our services - we'd like to supplement this with
   details from the access log analysis.

Yes, some of the data above would involve simple email reports to various
stakeholders. Other metrics are values we want to display on our
website, and others still are simply numbers that aren't for display, but
rather would be fed into various automation scripts.

The challenge is that while your information is valuable, many of our
use-cases require "joining" your data with other information - this is why
the HTML format reports aren't very useful to us. An example might be the
SLA reports we hope to create. Your application knows valuable things like
the number of hits by URL, response times and errors - but you don't know
things like network and server outages. Our goal would be to combine your
information with data we get from other tools to create the view we want.

We've looked at your JSON output and absent text reports, the best approach
we've found would be writing a little code to produce a text report using
your JSON file as input. This isn't a horrible solution, but it seems like
our requirements aren't that unusual, so we thought it was worth suggesting
as a feature for some future release.

Thanks again for your help!

Vince

**Reply by kevincobain2000** [reply]
I am working to solve same issue as you mentioned. For me, I have found that JSON is better than text as the post processing will be required at some extent for the application. One fix template to the text isn’t going to solve it for all. And for some sort of post processing, json is enough. 

For the intervals/monitoring and reporting SLA, we have multiple egreps with different parameters for same set of access_logs, suiting requirements of how report needs to be presented. 
These were how I thought as I faced same issue as yours.

**Comment by tdgvrr** [comment]
@kevincobain2000 - thanks for your reply. Lacking anything better, we ended up writing a little Java code to rip through the JSON file, pretty much as you suggested. Input parameters tell us what section of the JSON to format and return, and then we're able to build out the reports we want from these fragments.
